### PR TITLE
Use Sets, code cleanup

### PR DESCRIPTION
**Pull request by nijotz**
## Proposed Release Notes
* Use Sets, code cleanup

## Links
#70

### CODE REVIEW

File: bin/version-manager.js [hunk 1]
```diff
@@ -114,16 +114,18 @@ function resolveGlobs(globs, cb) {
...
-    dirs[path.dirname(p)] = true
-    return dirs
-  }, {}))
 
-  var mode = cmd.major ? 'major' : cmd.patch ? 'patch' : 'minor'
+  let directories = new Set()
+  filePaths.forEach((filePath) => {
+    directories.add(path.dirname(filePath))
+  })
+  directories = Array.from(directories)
```
Comment (michaelgoin) [inline bin/version-manager.js:126]: Looks like we are using Set for de-duplication.

Minor: remapping the type of directories is a bit awkward. Prob my type-safe background kicking in.

File: lib/versioned/suite.js [hunk 2]
```diff
@@ -29,23 +29,19 @@ util.inherits(Suite, EventEmitter)
...
+      testPackage.tests.forEach((test) => {
+        let dependencies = Object.keys(test.dependencies)
+        dependencies.forEach((dep) => packages.add(dep))
       })
-    })
-    return pkgs
-  }, {}))
+    }
+  })
+  packages = Array.from(packages)
```
Comment (michaelgoin) [inline lib/versioned/suite.js:42]: yay descriptive names.

another minor comment on swapping variable type. prob doesn't really matter given contained to same function.

**Review by michaelgoin — APPROVED** [review]
Looks good.

Couple minor comments.

I don't know how in-depth the tests are in this project, so I might recommend running our agent test suites through this change if you haven't already.

File: bin/version-manager.js (hunk 1 above)
Comment (nijotz) [reply bin/version-manager.js:126]: Ya it was a bit awkward. I would've left it a Set, but later on some code called a function on it that would only work with arrays..

Comment (michaelgoin) [reply bin/version-manager.js:126]: yeah. mostly i just don't like change types on an individual variable vs introducing a new variable so usage is consistent. but minor given the isolation of this usage.

**Pull request merged**